### ISSUE TITLE
Unexpected failure to verify @EnsuresCalledMethods post-condition on field

### ISSUE THREAD

**Issue by msridhar** [typetools/checker-framework#4279]
The following example is verified by the Called Methods Checker, as expected:

```java
import org.checkerframework.checker.calledmethods.qual.*;
class Test {
    static class Foo {
        void bar() {}
    }

    Foo foo1;
    Foo foo2;

    @EnsuresCalledMethods(value = "#1", methods = "bar")
    void callBar(Foo f) {
        f.bar();
    }

    @EnsuresCalledMethods(value = "this.foo1", methods = "bar")
    void barOnField() {
        callBar(this.foo1);
    }
}
```

But, an unexpected error is reported for this version:

```java
import org.checkerframework.checker.calledmethods.qual.*;
class Test {
    static class Foo {
        void bar() {}
    }

    Foo foo1;
    Foo foo2;

    @EnsuresCalledMethods(value = "#1", methods = "bar")
    void callBar(Foo f) {
        f.bar();
    }

    @EnsuresCalledMethods(value = "this.foo1", methods = "bar")
    void barOnField() {
        callBar(this.foo1);
        callBar(this.foo2); // just added this (irrelevant?) call
    }
}
```

The reported error is:
```
$ ~/git-repos/checker-framework/checker/bin/javac -processor CalledMethods  Test.java
Test.java:16: error: [contracts.postcondition.not.satisfied] postcondition of barOnField is not satisfied.
    void barOnField() {
         ^
  found   : no information about this.foo1
  required: this.foo1 is @CalledMethods("bar")
1 error
```

This seems like a false positive; I don't see how adding the call above could cause the postcondition to not hold.

**Comment by msridhar** [comment]
Never mind, this is correct behavior, since the second call might reset the state of `this.foo1`.
